File: instances/Biome.ts
import { Point2D } from "./GameTypes";

export enum Biome {
  NORMAL = 1,
  BARREN = 2,
  ASTEROIDFIELD = 3,
  NEBULA = 4,
  BLACKHOLE = 5,
  INVALID = 6
};

export function biomeToString(b: Biome) {
  switch(b) {
    case Biome.NORMAL:
      return "normal";
    case Biome.BARREN:
      return "barren";
    case Biome.ASTEROIDFIELD:
      return "asteroid field";
    case Biome.NEBULA:
      return "nebula";
    case Biome.BLACKHOLE:
      return "blackhole";
    case Biome.INVALID:
    default:
      return "invalid";
  }
}

export interface BiomeInfo {
  chunk: Point2D;
  biome: Biome;
}

File: cpp/src/Biome.cpp
#include <Biome.hpp>

namespace vasteroids {

static std::string strings[6] = {
  "normal",
  "barren",
  "asteroidfield",
  "nebula",
  "blackhole",
  "invalid"
};


const std::string& BiomeToString(const Biome& b) {
  return strings[static_cast<int>(b) - 1];
}
Napi::String BiomeToString(const Biome& b, const Napi::Env& env) {
  return Napi::String::New(env, BiomeToString(b));
}

}